### PR TITLE
Fix argument documentation

### PR DESCRIPTION
**Pull request by sgugger**
This decay is not weight decay, so should be renamed to not be confusing.

### CODE REVIEW

File: Sources/TensorFlow/Optimizers/SGD.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ public class SGD<Model: Differentiable>: Optimizer
     /// The momentum factor. It accelerates stochastic gradient descent in the relevant direction
     /// and dampens oscillations.
     public var momentum: Float
-    /// The weight decay.
+    /// The learning rate decay.
```
Comment (dan-zheng) [inline Sources/TensorFlow/Optimizers/SGD.swift]: In `Sources/TensorFlow/Optimizers/MomentumBased.swift`, there's another occurrence of `/// The weight decay.` that should be renamed to `/// The learning rate decay.`, if you'd like to update that as well!

Comment (sgugger) [reply Sources/TensorFlow/Optimizers/SGD.swift]: Changed this one as well.

**Comment by sgugger** [comment]
Also changed the precondition failure messages.
